src/Agent.Worker/Handlers/ProcessHandler.cs

```diff
@@ -217,7 +222,11 @@ private void OnOutputDataReceived(object sender, ProcessDataReceivedEventArgs e)
                         if (index > 0)
                         {
                             string key = line.Substring(0, index);
-                            string value = line.Substring(index + 1);
+
+                            if (_readonlyEnvVariables.Contains(key, StringComparer.OrdinalIgnoreCase))
+                            {
```

Comment:
Consider: use feature flag for this change + collect metrics to determine impact and effectiveness